buildpipeline/perf-pipeline.groovy

```diff
@@ -6,9 +6,374 @@
+        }
+
+        String pgoTestFlag = ((pgo == 'nopgo') ? '-nopgo' : '')
+
+        dir ('.') {
+            unstash "linux-${arch}-${pgo}${baselineString}-build-artifacts"
+            unstash "benchview-tools"
+            unstash "throughput-benchmarks-${arch}"
+            unstash "metadata"
+        }
+
+        // We want to use the baseline metadata for baseline runs. We expect to find the submission metadata in
+        // submission-metadata.py
+        if (isBaseline) {
+            sh "mv -f submission-metadata-baseline.json submission-metadata.json"
+        }
+
+        sh "./tests/scripts/perf-prep.sh --throughput"
+        sh "./init-tools.sh"
+        sh "python3 ./tests/scripts/run-throughput-perf.py -arch \"${arch}\" -os \"${os}\" -configuration \"${config}\" -opt_level ${optLevel} ${pgoTestFlag} -clr_root \"\${WORKSPACE}\" -assembly_root \"\${WORKSPACE}/${arch}ThroughputBenchmarks/lib\" -run_type \"${runType}\"  -benchview_path \"\${WORKSPACE}/tests/scripts/Microsoft.BenchView.JSONFormat/tools\""
+        archiveArtifacts allowEmptyArchive: false, artifacts:'throughput-*.csv,machinedata.json'
+    }
+}
+
+//-------------------------- Job Definitions --------------------------//
+
+String config = "Release"
+String runType = isPR() ? 'private' : 'rolling'
+
+String uploadString = '-uploadToBenchview'
+
+stage ('Get Metadata and download Throughput Benchmarks') {
     simpleNode('Windows_NT', '20170427-elevated') {
         checkout scm
+        String commit = getCommit()
+        def benchViewName = isPR() ? "coreclr private %ghprbPullTitle%" : "coreclr rolling %GIT_BRANCH_WITHOUT_ORIGIN% ${commit}"
+        def benchViewUser = getUserEmail()
+        bat "mkdir tools\n" +
+            "powershell Invoke-WebRequest https://dist.nuget.org/win-x86-commandline/v4.1.0/nuget.exe -OutFile %WORKSPACE%\\tools\\nuget.exe"
+        bat "%WORKSPACE%\\tools\\nuget.exe install Microsoft.BenchView.JSONFormat -Source http://benchviewtestfeed.azurewebsites.net/nuget -Prerelease -ExcludeVersion"
+        bat "%WORKSPACE%\\tools\\nuget.exe install Microsoft.BenchView.ThroughputBenchmarks.x64.Windows_NT -Source https://dotnet.myget.org/F/dotnet-core -Prerelease -ExcludeVersion"
+        bat "%WORKSPACE%\\tools\\nuget.exe install Microsoft.BenchView.ThroughputBenchmarks.x86.Windows_NT -Source https://dotnet.myget.org/F/dotnet-core -Prerelease -ExcludeVersion"
+        bat "set \"GIT_BRANCH_WITHOUT_ORIGIN=%GitBranchOrCommit:*/=%\"\n" +
+            "py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"${benchViewName}\" --user-email \"${benchViewUser}\"\n" +
+            "py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\build.py\" git --branch %GIT_BRANCH_WITHOUT_ORIGIN% --type ${runType}\n" +
+            "py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"${benchViewName}-baseline\" --user-email \"${benchViewUser}\" -o submission-metadata-baseline.json\n"
```

Comment:
>"py \"%WORKSPACE%\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"${benchViewName}-baseline\" --user-email \"${benchViewUser}\" -o submission-metadata-baseline.json\n" [](start = 12, length = 193)

I think we should have one submission-metadata for both base & diff. A submission can have two jobs with different builds. Optionally, two submission-metadata with the same cuid.

Reply:
How would the user know which was the baseline run and which was the real run?